test/python/apps/solana.py

```diff
@@ -0,0 +1,145 @@
+from typing import List, Generator
+from enum import IntEnum
+from contextlib import contextmanager
+
+from ragger.backend.interface import BackendInterface, RAPDU
+
+
+class INS(IntEnum):
+    # DEPRECATED - Use non "16" suffixed variants below
+    INS_GET_APP_CONFIGURATION16 = 0x01
+    INS_GET_PUBKEY16 = 0x02
+    INS_SIGN_MESSAGE16 = 0x03
+    # END DEPRECATED
+    INS_GET_APP_CONFIGURATION = 0x04
+    INS_GET_PUBKEY = 0x05
+    INS_SIGN_MESSAGE = 0x06
+    INS_SIGN_OFFCHAIN_MESSAGE = 0x07
+
+
+CLA = 0xE0
+
+P1_NON_CONFIRM = 0x00
+P1_CONFIRM = 0x01
+
+P2_NONE = 0x00
+P2_EXTEND = 0x01
+P2_MORE = 0x02
+
+PUBLIC_KEY_LENGTH = 32
+
+MAX_CHUNK_SIZE = 255
+
+STATUS_OK = 0x9000
+
+
+class ErrorType:
+    NO_APP_RESPONSE = 0x6700
+    SDK_EXCEPTION = 0x6801
+    SDK_INVALID_PARAMETER = 0x6802
+    SDK_EXCEPTION_OVERFLOW = 0x6803
+    SDK_EXCEPTION_SECURITY = 0x6804
+    SDK_INVALID_CRC = 0x6805
+    SDK_INVALID_CHECKSUM = 0x6806
+    SDK_INVALID_COUNTER = 0x6807
+    SDK_NOT_SUPPORTED = 0x6808
+    SDK_INVALID_STATE = 0x6809
+    SDK_TIMEOUT = 0x6810
+    SDK_EXCEPTION_PIC = 0x6811
+    SDK_EXCEPTION_APP_EXIT = 0x6812
+    SDK_EXCEPTION_IO_OVERFLOW = 0x6813
+    SDK_EXCEPTION_IO_HEADER = 0x6814
+    SDK_EXCEPTION_IO_STATE = 0x6815
+    SDK_EXCEPTION_IO_RESET = 0x6816
+    SDK_EXCEPTION_CX_PORT = 0x6817
+    SDK_EXCEPTION_SYSTEM = 0x6818
+    SDK_NOT_ENOUGH_SPACE = 0x6819
+    NO_APDU_RECEIVED = 0x6982
+    USER_CANCEL = 0x6985
+    SOLANA_INVALID_MESSAGE = 0x6a80
+    SOLANA_SUMMARY_FINALIZE_FAILED = 0x6f00
+    SOLANA_SUMMARY_UPDATE_FAILED = 0x6f01
+    UNIMPLEMENTED_INSTRUCTION = 0x6d00
+    INVALID_CLA = 0x6e00
+
+
+def _extend_and_serialize_multiple_derivations_paths(derivations_paths: List[bytes]):
+    serialized: bytes = len(derivations_paths).to_bytes(1, byteorder='little')
+    for derivations_path in derivations_paths:
+        serialized += derivations_path
+    return serialized
+
+
+class SolanaClient:
+    client: BackendInterface
+
+    def __init__(self, client):
```

Comment:
`client: BackendInterface`

Reply:
I don't understand ?

Reply:
```python
def __init__(self, client: BackendInterface):
```

So that `mypy` would yield against the constructor of `SolanaClient` rather than its `client` class attribute.

Reply:
Done